modules/bim/game/main/include/bim/game/level_generation.hpp

```diff
@@ -12,9 +12,11 @@ namespace bim::game
 
   constexpr std::uint8_t g_flame_power_up_count_in_level = 8;
   constexpr std::uint8_t g_bomb_power_up_count_in_level = 10;
+  constexpr std::uint8_t g_invisibility_power_up_count_in_level = 6;
 
   void generate_basic_level_structure(arena& arena);
   void insert_random_brick_walls(arena& arena, entt::registry& registry,
                                  random_generator& random_generator,
-                                 std::uint8_t brick_wall_probability);
+                                 std::uint8_t brick_wall_probability,
+                                 bool invisibility_power_up_enabled = false);
```

Comment:
You can pass the `feature_flags directly` here, I think it will be better than having a list of booleans on the long run, even if some flags are not handled in this function.